src/transform.rs

```diff
@@ -69,21 +67,50 @@ pub struct Decomposed<V: VectorSpace, R> {
     pub disp: V,
 }
 
-impl<P: EuclideanSpace, R: Rotation<P>> Transform<P> for Decomposed<P::Diff, R>
+// using `EuclideanRotation` here to avoid the possibility that
+// the same struct implements `Rotation<P1>` and `Rotation<P2>`
+// and the compiler has to guess what `P` is.
+impl<P: EuclideanSpace, R: EuclideanRotation<Euclidean = P>> One for Decomposed<P::Diff, R>
```

Comment:
should this be `impl<R: EuclideanRotation> One for Decomposed<<R::Euclidean as Xxx>>::Diff, R>`?

Reply:
It could be that, but that would make all mentions of `P` a lot more verbose, and there are several of them.